sympy/functions/combinatorial/numbers.py

```diff
@@ -2175,6 +2175,155 @@ def eval(cls, n):
+    SW-NE diagonal. The nth Schroeder number is the number of Schroeder
+    paths of length n. The Schroeder numbers were named after the German
+    mathematician Ernst Schroeder.
+
+    Schroeder numbers are the integer sequences defined by the initial
+    terms `S_0 = 1`, `S_1 = 2` and the two-term recurrence relation `S_n
+    = \frac{6*n - 3}{n + 1} * S_{n-1} - \frac{n - 2}{n + 1} * S_{n-2}`.
+
+
+    Examples
+    ========
+
+    >>> from sympy.functions.combinatorial.numbers import schroeder
+
+    Compute the nth Schroeder number:
+
+    >>> schroeder(2)
+    2
+    >>> schroeder(5)
+    90
+
+    Check if a number is a Schroeder number:
+
+    >>> schroeder.is_schroeder(2)
+    True
+    >>> schroeder.is_schroeder(7)
+    False
+
+    Find the first n Schroeder numbers:
+
+    >>> schroeder.find_first_n_schroeder_numbers(1)
+    [1]
+    >>> schroeder.find_first_n_schroeder_numbers(10)
+    [1, 2, 6, 22, 90, 394, 1806, 8558, 41586, 206098]
+
+    Find Schroeder numbers in a given range:
+
+    >>> schroeder.find_schroeder_numbers_in_range(1, 10)
+    [1, 2, 6]
+    >>> schroeder.find_schroeder_numbers_in_range(2, 400)
+    [2, 6, 22, 90, 394]
+
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Schr%C3%B6der_number
+    .. [2] https://mathworld.wolfram.com/SchroederNumber.html
+
+    """
+
+    @staticmethod
+    def is_schroeder(n):
+        try:
+            n = as_int(n)
+        except ValueError:
+            False
+        if n > 0:
+            if n in (1, 2):
+                return True
+
+            sn1 = 1
+            sn = 2
+            i = 2
+            while sn < n:
+                b = ((6*i - 3)*sn - (i - 2)*sn1) // (i + 1)
+                sn1 = sn
+                sn = b
+                i = i+1
+
+            return sn == n
+
+        else:
+            return False
+
+    @staticmethod
+    def find_first_n_schroeder_numbers(n):
+        try:
+            n = as_int(n)
+        except ValueError:
+            raise ValueError('The provided number must be a positive integer')
+        if n < 0:
+            raise ValueError('The provided number must be a positive integer')
+        if n == 1:
+            return [1]
+        schroder = [1]
+        itr = schroeder._iterator()
+        for _ in range(1, n):
+            schroder.append(next(itr))
+
+        return schroder
+
+    @staticmethod
+    def find_schroeder_numbers_in_range(x, y):
+        """
+        Returns all the schroeder numbers in the range (x, y), x and y
+        inclusive.
+
+        """
+
+        if x < 0 or y < x:
+            raise ValueError('The provided range is not valid. This condition should satisfy x <= y')
+        else:
+            schroder = []
+            if x <= 1 <= y:
+                schroder.append(1)
+            itr = schroeder._iterator()
+            for _ in range(1, y+1):
+                t = next(itr)
+                if t >= x and t <= y:
+                    schroder.append(t)
```

Comment:
Rather than calling next the natural thing to do here would be to loop over the iterator and use break to exit the loop.